src/Visualization/Python/GenerateTetrahedralConnectivity.py

```diff
@@ -0,0 +1,208 @@
+                " Available subfiles: "
+                + str(available_subfiles(h5file, extension=".vol"))
+            ) from err
+
+        # Sort timesteps by time
+        temporal_ids_and_values = sorted(
+            [
+                (key, vol_subfile[key].attrs["observation_value"])
+                for key in vol_subfile.keys()
+            ],
+            key=lambda key_and_time: key_and_time[1],
+        )
+
+        # Stride through timesteps
+        for temporal_id, time in temporal_ids_and_values[::stride]:
+            # Filter by start and end time
+            if start_time is not None and time < start_time:
+                continue
+            if stop_time is not None and time > stop_time:
+                break
+
+            data_at_id = vol_subfile[temporal_id]
+
+            if delete_tetrahedral_connectivity:
+                if "tetrahedral_connectivity" in data_at_id:
+                    del data_at_id["tetrahedral_connectivity"]
+                continue
+
+            x_coords = np.asarray(data_at_id[coordinates + "_x"])
+            y_coords = np.asarray(data_at_id[coordinates + "_y"])
+            if (coordinates + "_z") in data_at_id:
+                z_coords = np.asarray(data_at_id[coordinates + "_z"])
+                coords = np.column_stack((x_coords, y_coords, z_coords))
+            else:
+                coords = np.column_stack((x_coords, y_coords))
+
+            logger.info(
+                f"Generating tetrahedral connectivity at {temporal_id}/{time}. "
+                "This may take a few minutes and may even fail depending on "
+                "the grid structure."
+            )
+            delaunay = spatial.Delaunay(coords)
+            data_at_id.create_dataset(
+                "tetrahedral_connectivity", data=delaunay.simplices.flatten()
+            )
+
+    for h5file in h5files:
+        h5file[0].close()
+
+
+@click.command(
+    name="generate-tetrahedral-connectivity",
+    help=generate_tetrahedral_connectivity.__doc__,
+)
+@click.argument(
+    "h5files",
+    type=click.Path(exists=True, file_okay=True, dir_okay=False, readable=True),
+    nargs=-1,
+    required=True,
+)
+@click.option(
+    "--subfile-name",
+    "-d",
+    help=(
+        "Name of the volume data subfile in the H5 files. A '.vol' extension is"
+        " added if needed. If unspecified, and the first H5 file contains only"
+        " a single '.vol' subfile, choose that. Otherwise, list all '.vol'"
+        " subfiles and exit."
+    ),
+)
+@click.option(
+    "--stride", default=1, type=int, help="View only every stride'th time step"
+)
+@click.option(
+    "--start-time",
+    type=float,
+    help=(
+        "The earliest time at which to start visualizing. The start-time "
+        "value is included."
+    ),
+)
+@click.option(
+    "--stop-time",
+    type=float,
+    help=(
+        "The time at which to stop visualizing. The stop-time value is "
+        "not included."
```

Comment:
Actually I think it _is_ included, no? This might be a documentation bug in GenerateXdmf.py as well.